ffi/capi/src/week.rs

```diff
@@ -14,127 +14,71 @@ pub mod ffi {
-            Box::new(WeekCalculator(calculator))
-        }
 
         /// Returns the weekday that starts the week for this object's locale
-        #[diplomat::rust_link(icu::calendar::week::WeekCalculator::first_weekday, StructField)]
+        #[diplomat::rust_link(icu::calendar::week::WeekInformation::first_weekday, StructField)]
         #[diplomat::attr(auto, getter)]
         pub fn first_weekday(&self) -> Weekday {
             self.0.first_weekday.into()
         }
-        /// The minimum number of days overlapping a year required for a week to be
-        /// considered part of that year
-        #[diplomat::rust_link(icu::calendar::week::WeekCalculator::min_week_days, StructField)]
-        #[diplomat::attr(auto, getter)]
-        pub fn min_week_days(&self) -> u8 {
-            self.0.min_week_days
+
+        #[diplomat::rust_link(icu::calendar::week::WeekInformation::weekend, StructField)]
+        #[diplomat::rust_link(icu::calendar::week::WeekendSet::contains, FnInStruct)]
+        pub fn is_weekend(&self, day: Weekday) -> bool {
+            self.0.weekend.contains(day.into())
         }
 
-        #[diplomat::rust_link(icu::calendar::week::WeekCalculator::weekend, FnInStruct)]
+        #[diplomat::rust_link(icu::calendar::week::WeekInformation::weekend, FnInStruct)]
         #[diplomat::attr(auto, getter)]
-        pub fn weekend(&self) -> WeekendContainsDay {
-            let mut contains = WeekendContainsDay::default();
-            for day in self.0.weekend() {
-                match day {
-                    icu_calendar::types::Weekday::Monday => contains.monday = true,
-                    icu_calendar::types::Weekday::Tuesday => contains.tuesday = true,
-                    icu_calendar::types::Weekday::Wednesday => contains.wednesday = true,
-                    icu_calendar::types::Weekday::Thursday => contains.thursday = true,
-                    icu_calendar::types::Weekday::Friday => contains.friday = true,
-                    icu_calendar::types::Weekday::Saturday => contains.saturday = true,
-                    icu_calendar::types::Weekday::Sunday => contains.sunday = true,
-                }
-            }
-            contains
+        pub fn weekend(&self) -> Box<WeekdaySetIterator> {
+            Box::new(WeekdaySetIterator(self.0.weekend()))
         }
     }
 
     /// Documents which days of the week are considered to be a part of the weekend
-    #[diplomat::rust_link(icu::calendar::week::WeekCalculator::weekend, FnInStruct)]
-    #[derive(Default)]
-    pub struct WeekendContainsDay {
-        pub monday: bool,
-        pub tuesday: bool,
-        pub wednesday: bool,
-        pub thursday: bool,
-        pub friday: bool,
-        pub saturday: bool,
-        pub sunday: bool,
-    }
-}
+    #[diplomat::rust_link(icu::calendar::week::WeekdaySetIterator, Struct)]
+    #[diplomat::opaque]
+    pub struct WeekdaySetIterator(icu_calendar::week::WeekdaySetIterator);
```

Comment:
Question/Issue: Why did you replace the no-alloc `WeekendContainsDay` with an opaque, allocating `WeekdaySetIterator`. I vaguely remember a conversation the team had about this when it was first added and I recall us picking this design in order to reduce allocations.

Reply:
Because it's a weird API. But I'll revert

Reply:
actually I also added an `is_weekend` API, which does what the previous API did. It's nice to also have the iterator, to not have to write a 7-branch if statement.

Reply:
A weekend day iterator is not a great API because weekend days are not necessarily contiguous.

I'd be fine with only exporting `is_weekend` instead of the struct.

Reply:
What is the use case for "first weekend day"? That seems like a footgun if anything. I think the use case of this API is so that you can display weekend days with a different background color in a grid layout, for example.

Reply:
ok: #6281

This comment thread still tracks the concern I had about adding an allocation in FFI where there wasn't previously an allocation, a regression compared to 2.0.0-beta1. I would rather you split that one piece into a different PR in order to reduce the amount of disruption on the main branch.

Reply:
I won't

Reply:
I don't understand why you don't want to decouple these two changes (nothing in the PR or issue has anything to do with weekend iteration). But then I don't see a way around blocking this on a WG discussion, which we can have tomorrow.

Reply:
This PR redid the WeekInformation types in Rust and FFI.

I don't want to decouple it because it's extra work for me. Aligning the FFI API of this type that significantly changed in this PR is not a crime.